google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -143,175 +153,131 @@ def create(
                 labels specified inside index file refenced by
                 [import_schema_uri][google.cloud.aiplatform.v1beta1.ImportDataConfig.import_schema_uri],
                 e.g. jsonl file.
-            project: Optional[str]=None,
+            project (Optional[str]):
                 Project to upload this model to. Overrides project set in
                 aiplatform.init.
-            location: Optional[str]=None,
+            location (Optional[str]):
                 Location to upload this model to. Overrides location set in
                 aiplatform.init.
-            credentials: Optional[auth_credentials.Credentials]=None,
+            credentials (Optional[auth_credentials.Credentials]):
                 Custom credentials to use to upload this model. Overrides
                 credentials set in aiplatform.init.
+            request_metadata (Optional[Sequence[Tuple[str, str]]]):
+                Strings which should be sent along with the request as metadata.
             sync (bool):
                 Whether to execute this method synchronously. If False, this method
                 will be executed in concurrent Future and any downstream object will
                 be immediately returned and synced when the Future has completed.
+
         Returns:
             dataset (Dataset):
                 Instantiated representation of the managed dataset resource.
-        """
-        utils.validate_display_name(display_name)
 
-        is_tabular_dataset_metadata = (
-            metadata_schema_uri == schema.dataset.metadata.tabular
-        )
+        """
 
-        if bool(bq_source) and not is_tabular_dataset_metadata:
-            raise ValueError(
-                "A tabular metadata_schema uri must be used when using a BigQuery source"
-            )
+        utils.validate_display_name(display_name)
 
         api_client = cls._instantiate_client(location=location, credentials=credentials)
 
-        # If this is tabular enrich the dataset metadata with source
-        dataset_metadata = {}
-        if is_tabular_dataset_metadata:
-            if gcs_source:
-                dataset_metadata = {"input_config": {"gcs_source": {"uri": gcs_source}}}
-            elif bq_source:
-                dataset_metadata = {
-                    "input_config": {"bigquery_source": {"uri": bq_source}}
-                }
+        if metadata_schema_uri == schema.dataset.metadata.tabular:
```

Comment:
I wonder if we should push this conditional instantiation of correct Datasource class as a helper method in the _datasource module.